MobileBuy/sample/src/main/java/com/shopify/sample/activity/CheckoutActivity.java

```diff
@@ -53,7 +54,15 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         setTitle(R.string.checkout);
-        setContentView(R.layout.checkout_activity);
+
+        OptimizelyClient optimizelyClient = getSampleApplication().getOptimizelyManager().getOptimizely();
+        Variation product_sorting_variation = optimizelyClient.getVariation("checkout_activity_layout",
+                getSampleApplication().getUser());
+        if (product_sorting_variation.getKey().equals("include_fake_text")) {
```

Comment:
Good point @onufryk. @elliotykim we should also call `getVariation` with an invalid experiment key and make sure that we get null. In the case we get null we should have a default case, and in this implementation you have it would be a new activity. If not then maybe we can just implement the other option that @wangjoshuah mentioned, where we only change the text of the button.